enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationService.java

```diff
@@ -376,40 +352,85 @@ public void processDocumentVerificationResult(final OwnerId ownerId, final Ident
      * Move identity verification to {@code DOCUMENT_VERIFICATION} phase and status based on the given document verifications.
      *
      * @param idVerification Identity verification entity.
-     * @param docVerifications Document verifications to determine identity verification status.
+     * @param docVerificationsToProcess Document verifications to determine identity verification status.
      */
     private void moveToDocumentVerificationAndStatusByDocuments(
             final IdentityVerificationEntity idVerification,
-            final List<DocumentVerificationEntity> docVerifications,
+            final List<DocumentVerificationEntity> docVerificationsToProcess,
             final OwnerId ownerId) {
 
-        final IdentityVerificationPhase phase = IdentityVerificationPhase.DOCUMENT_VERIFICATION;
-        final Date now = ownerId.getTimestamp();
-        if (docVerifications.stream()
+        final String identityVerificationId = idVerification.getId();
+        // docVerificationsToProcess have been modified, so idVerification.getDocumentVerifications() must be reloaded to reflect these modifications
```

Comment:
👀 This was a little bit tricky. Status of documents is change at `checkVerificationResult`, but these changes were not yet available in the 1st level cache to get them via `idVerification.getDocumentVerifications()`.
`allDocumentVerifications` represents all documents, `docVerificationsToProcess` just current step.